nexus/src/external_api/http_entrypoints.rs

```diff
@@ -5677,11 +5677,14 @@ impl NexusExternalApi for NexusExternalApiImpl {
             };
             let instance_lookup =
                 nexus.instance_lookup(&opctx, instance_selector)?;
+            let params::EphemeralIpCreate { pool, ip_version } =
+                ip_to_create.into_inner();
             let ip = nexus
                 .instance_attach_ephemeral_ip(
                     &opctx,
                     &instance_lookup,
-                    ip_to_create.into_inner().pool,
+                    pool,
+                    ip_version,
```

Comment:
It seems like somewhere we want to fail if you provide an IP version and a pool, and the version you gave doesn't match the version of the pool. Is that the case?

Reply:
I need to add this match check, good point.

Reply:
Moving this up to communicate the error before the saga starts. After that, ready-for-re-review.